src/client/datascience/dataScienceSurveyBanner.ts

```diff
@@ -122,11 +122,13 @@ export class DataScienceSurveyBanner implements IJupyterExtensionBanner {
         switch (response) {
             case this.bannerLabels[DSSurveyLabelIndex.Yes]: {
                 await this.launchSurvey();
-                await this.disable();
+                // Disable for 6 months
+                await this.disable(6);
                 break;
             }
             case this.bannerLabels[DSSurveyLabelIndex.No]: {
-                await this.disable();
+                // Disable for 3 months
+                await this.disable(3);
```

Comment:
I think that you implemented this fine. I'm just a bit surprised at the requirement. Do we really want to say no to a survey, and then prompt them again in 3 months? I would think at a minimum we would need a never again option for this to be legit.

Reply:
yes